rover_video/src/screenshot_server.hpp

```diff
@@ -0,0 +1,681 @@
+        posID += 4;
+        nextDotPos = cameraURL.find(':', posID);
+
+        if (nextDotPos != std::string::npos)
+        {
+            camID = cameraURL.substr(posID, nextDotPos - posID);
+        }
+        else
+        {
+            camID = cameraURL.substr(posID);
+        }
+    }
+    else
+    {
+        RCLCPP_ERROR(LOGGER, "'144.' not found.");
+    }
+
+    RCLCPP_DEBUG(LOGGER, "Camera ID: %s", camID.c_str());
+
+    return camID;
+}
+
+// get current time in ISO format for filename
+std::string CameraNode::getCurrentTime()
+{
+    std::chrono::time_point<std::chrono::system_clock> now = std::chrono::system_clock::now();  // get system time
+
+    std::time_t now_time = std::chrono::system_clock::to_time_t(now);  // convert to real time
+
+    std::tm tm_now = *std::localtime(&now_time);  // convert to calendar time
+
+    std::stringstream current_time_output;
+    current_time_output << std::put_time(&tm_now, "%FT%T");  // ISO 8601 format
+
+    return current_time_output.str();
+}
+
+std::string CameraNode::getFileName(const std::string capture_name, std::string camURL, int state)
+{
+    std::string filename;
+
+    std::string time = getCurrentTime();
+    std::string latitude = std::to_string(last_latitude);
+    std::string longitude = std::to_string(last_longitude);
+    std::string ID = getCamID(camURL);
+
+    switch (state)
+    {
+        case SCREENSHOT:
+            filename = capture_name.empty()
+                           ? time + "_lat:" + latitude + "_long:" + longitude + "_" + ID + "_screenshot.png"
+                           : time + "_lat:" + latitude + "_long:" + longitude + "_camID:" + ID + "_" + capture_name;
+            // Example : 2024-12-10T20:50:00_GPS_30_screenshot.png
+            break;
+
+        case VIDEO:
+            filename = capture_name.empty()
+                           ? time + "_lat:" + latitude + "_long:" + longitude + "_" + ID + "_recording.avi"
+                           : time + "_lat:" + latitude + "_long:" + longitude + "_camID:" + ID + "_" + capture_name;
+            // Example : 2024-12-10T20:50:00_GPS_30_recording.avi
+    }
+    return filename;
+}
+
+const std::string CameraNode::getFolderPath(int state)
+{
+    std::string folderPath;
+
+    std::string dir = GET_PACKAGE_SOURCE_DIR("rover_video");  // finds the path to our package
+
+    switch (state)
+    {
+        case SCREENSHOT:
+            // Path necessary for the screenshots folder
+            folderPath = std::string(dir) + "/src/screenshots";
+            break;
+
+        case VIDEO:
+            // Path necessary for the recordings folder
+            folderPath = std::string(dir) + "/src/recordings";
+            break;
+    }
+
+    return folderPath;
+}
+
+// Verifies if screenshot folder already exists
+bool CameraNode::folderExists(const std::string& path)
+{
+    struct stat info;
+    return (stat(path.c_str(), &info) == 0 && (info.st_mode & S_IFDIR));  // I dont exactly understand this part
+}
```

Comment:
Here's an example of stat usage for your case. You'll figure that it needs to handle more than 2 possibilities so a enum class return type might be necessary. Modifying the flow to just be a check if there's a valid folder instead would let you use bool instead.

```suggestion
bool CameraNode::folderExists(const std::string& path)
{
    struct stat fileInfo;
    if (stat(path.c_str(), &fileInfo) != 0) // Successfully found information of element with this path
    {
        // Nothing there or invalid permission (see stat documentation)
        return false;
    }
    
    if (fileInfo.st_mode & S_IFDIR) // Check if file type is folder
    {
        return true;
    }
    else
    {
        #error Element already exist with this path and name but ins't a folder
        // Todo return; Flow will need to change a bit to handle this case
    }
}
```